model-view-intent/src/main/java/com/iluwatar/model/view/intent/actions/CalculatorAction.java

```diff
@@ -0,0 +1,15 @@
+package com.iluwatar.model.view.intent.actions;
+
+/**
+ * Defines what outside interactions can be consumed by view model.
+ * */
+public interface CalculatorAction {
```

Comment:
I think you can use an enum-based implementation to get rid of the zillion little `CalculatorAction` derived classes

Reply:
I thought about it, but the main strength of the MVI pattern is the reduced API of the ViewModel. The only way to achieve it in a non brittle way is having an option to pass different amounts of arguments with different types to an Intent. In Kotlin it's straight thanks to sealed classes, as far as I'm aware, Java Enums don't support constructors with differing parameters, which would both significantly limit the use cases and hide MVIs main feature.